hyacinth/session.py

```diff
@@ -236,26 +241,103 @@ def delete_folder(self, id, **kwargs):
         url = Session.__make_url(f"folders/{id}")
         return self.__delete_resource(url, **kwargs)
 
-    def upload_document(self, name, parent_id, parent_type, document):
+    def upload_document(self, name, parent_id, parent_type, document, progress_update=lambda *args: None):
         """POST a new Document, PUT the data, and PATCH Document as fully_uploaded."""
+        file_size = os.path.getsize(document)
+        if file_size > PART_SIZE:
+            return self.upload_multipart_document(
+                name,
+                parent_id,
+                parent_type,
+                document,
+                progress_update,
+            )
+
+        with open(document, "rb") as f:
+            post_url = Session.__make_url("documents")
+            clio_document = self.__post_resource(
+                post_url,
+                params={"fields": "id,latest_document_version{uuid,put_url,put_headers}"},
+                json={
+                    "data": {"name": name, "parent": {"id": parent_id, "type": parent_type}}
+                },
+            )
+
+            put_url = clio_document["data"]["latest_document_version"]["put_url"]
+            put_headers = clio_document["data"]["latest_document_version"]["put_headers"]
+
+            headers_map = {}
+            for header in put_headers:
+                headers_map[header["name"]] = header["value"]
+
+            # We actually DON'T want to use the authenticated client here
+            requests.put(put_url, headers=headers_map, data=f, timeout=5)
```

Comment:
How come?

Reply:
We're making a request to Amazon S3 instead of the Clio; however, the `OAuth2` authenticated session is only valid for requests to Clio.